### PR TITLE
Add cancel job

### PR DESCRIPTION
**Pull request by RussTheAerialist**

### CODE REVIEW

File: samples/kill_all_jobs.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+####
+# This script demonstrates how to list all of the workbooks or datasources
```
Comment (shinchris) [inline samples/kill_all_jobs.py]: Description here and on line 15 needs update

File: samples/kill_all_jobs.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    logging.basicConfig(level=logging_level)
+
+    # SIGN IN
+    tableau_auth = TSC.TableauAuth(args.username, password, args.site)
+    server = TSC.Server(args.server, use_server_version=True)
+    with server.auth.sign_in(tableau_auth):
+        req = TSC.RequestOptions()
+
+        req.filter.add(TSC.Filter("progress", TSC.RequestOptions.Operator.LessThanOrEqual, 0))
+        for job in TSC.Pager(server.jobs.get, request_opts=req):
```
Comment (t8y8) [inline samples/kill_all_jobs.py]: Nit: You can just pass the endpoint

File: tableauserverclient/server/endpoint/endpoint.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def delete_request(self, url):
         # We don't return anything for a delete
         self._make_request(self.parent_srv.session.delete, url, auth_token=self.parent_srv.auth_token)
 
-    def put_request(self, url, xml_request, content_type='text/xml'):
+    def put_request(self, url, xml_request=None, content_type='text/xml'):
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/endpoint.py]: So cancel takes an empty (as in `b''`) PUT? Didn't run job take an empty `<tsRequest/ >`?

File: tableauserverclient/server/endpoint/jobs_endpoint.py [hunk 4]
```diff
@@ -17,12 +18,20 @@ def get(self, job_id=None, req_options=None):
             import warnings
             warnings.warn("Jobs.get(job_id) is deprecated, update code to use Jobs.get_by_id(job_id)")
             return self.get_by_id(job_id)
+        if isinstance(job_id, RequestOptionsBase):
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/jobs_endpoint.py]: I'm so sad. But customers will love the functionality!

File: tableauserverclient/server/endpoint/endpoint.py (hunk 3 above)
Comment (RussTheAerialist) [reply tableauserverclient/server/endpoint/endpoint.py]: yeah...

File: tableauserverclient/server/endpoint/jobs_endpoint.py (hunk 4 above)
Comment (RussTheAerialist) [reply tableauserverclient/server/endpoint/jobs_endpoint.py]: yeah once we hit 1.0, we can remove this.

**Pull request merged**